test/src/org/omegat/filters/INIFilterTest.java

```diff
@@ -50,6 +50,12 @@ public void testLoad() throws Exception {
         checkMulti("Value", "Section/ID", null, null, null, null);
         checkMulti("Value2", "Section/ID2", null, null, null, null);
         checkMulti("Value3", "Section/ID3", null, null, null, null);
+        checkMulti("Value4", "Section/ID4", null, null, null, null);
+        checkMulti("Continued Value4", "Section/#0#", null, null, null, null);
+        checkMulti("Value5", "Section/ID5", null, null, null, null);
+        checkMulti("Continued Value5 start with space", "Section/#1#", null, null, null, null);
```

Comment:
Sorry, I don't know why it took me so long to realize this, but:

1. Would it not make more sense for the value for `ID5` to be `Value5\n Continued Value5 start with space`? Are multiline values always really intended to be **collections of multiple values**? Or are they **single values spanning multiple lines**? If the latter, then we should extract as a single string and rely on segmentation to split them up.
2. If the answer to (1) is "no" (they are **collections** of values) then I think the keys for a collection should probably go like
  - `Section/ID5/0`
  - `Section/ID5/1`
  - `Section/ID5/2`

(How many levels of nesting can INI have? I see `[section]` and then the value ID. If that is the limit of nesting, then I think a `three/level/key` is sufficient to identify "virtual" values.)

Reply:
There is no concrete specification for multi-line value.
cf. https://en.wikipedia.org/wiki/INI_file

I don't know which is better, but (2) seems to be safe.

Reply:
I did (2) above, to the degree it's possible with the current implementation of the filter: https://github.com/omegat-org/omegat/pull/75/commits/f00ec0924a5d6484fc574c95c6e6000fd2acd3fd

Unfortunately at the time when we would build the key `Section/ID5/0` we don't yet know if there are continuation lines, so I output it as `Section/ID5` and then the first continuation line is `Section/ID5/1`.

It's also not feasible to implement (1) with the current implementation; we would have to build up values until we know there are no more continuation lines.

I would like to see an example of a real-world INI file that this improvement is targeting, so I can get a better feel for how important (1) is.

Reply:
One of targets of translation projects is a game translation. Some game project which use old version of Unreal Engine takes INI style file. (new version has a l10n feature which use po files)

eg. https://docs.unrealengine.com/udk/Three/LocalizedTextFiles.html

There seems no standard specification especially for multi line messages.

Here is a small example;

```
[System]

Hiscore = HighScore
Yes = Yes
No = no
Finished_All_Notification = Wow! It seems you have finished all the stages
You are awesome!
There will be more of them!
Congratulations = Congratulations

[UI]

EnableTranslation = You can edit the translation any of the available languages.
All you need to do is:
• browse to the localization folder
• Copy the localization file, of the specific language, you want to change
• Paste it to the folder


[Language]

English = English
Chinese = Chinese
Russian = Russian
German = German
French = French
Spanish = Spanish


```

Reply:
Thanks, that's helpful.

It seems pretty clear to me that the correct strategy is to parse as

- Key: `Finished_All_Notification`
- Value: `Wow! It seems you have finished all the stages\nYou are awesome!\nThere will be more of them!`

But I am not interested in rewriting the filter to support this.

What we have now is a compromise that at least is an improvement over the status quo. So I will squash and merge unless you are interested in working on it further.

Reply:
It is OK to merge it. Thank you.